src/components/AddRoomModal/AddRoomModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import {
+  StyledOpenBtn,
+  StyledModal,
+  StyledOverlay,
+  StyledModalContent,
+  StyledModalHeadline,
+  StyledLabel,
+  StyledAddButton,
+  StyledCloseModalBtn,
+  StyledInput,
+  StyledSelect,
+} from "./AddRoomModal.styled";
+import { RoomType } from "@prisma/client";
+import { formatRoomType } from "@/lib/formatRoomType";
+import { mutate } from "swr";
+
+export default function AddRoomModal() {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [name, setName] = useState<string | null>(null);
+  const [type, setType] = useState<RoomType | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+    setName("");
+    setType(null);
+    setError(null);
+  };
+
+  const handleAddRoom = () => {
+    if (!name || !type) {
+      setError("Please fill in all fields");
+      return;
+    }
+
+    fetch("/api/rooms", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        name,
+        type,
+      }),
+    }).then(() => {
+      mutate("/api/rooms");
+    });
+
+    toggleModal();
+    setName(null);
+    setType(null);
+    setError(null);
+  };
+
+  //TODO: <br> to margin
```

Comment:
```suggestion
  // TODO: <br> to margin
```

You can also add a rule to eslint to remind you to space out comments, just update your `.eslintrc` config file to match the following:


```json
{
  "extends": "next/core-web-vitals",
  "rules": [
    "spaced-comment": "error"
  ]
}
```

Reply:
Perfect, that very helpful :)

Reply:
✅